kazu/training/modelling_utils.py

```diff
@@ -64,6 +79,51 @@ def get_label_list_from_model(model_config_path: PathLike) -> list[str]:
     return label_list
 
 
+class LSManagerViewWrapper:
+    def __init__(self, view: LabelStudioAnnotationView, ls_manager: LabelStudioManager):
+        self.ls_manager = ls_manager
+        self.view = view
+
+    def get_gold_ents_for_side_by_side_view(self, docs: list[Document]) -> list[list[Document]]:
+        result = []
+        for doc in docs:
+            doc_cp = copy.deepcopy(doc)
+            if PROCESSING_EXCEPTION in doc_cp.metadata:
+                logger.error(doc.metadata[PROCESSING_EXCEPTION])
+                break
+            for section in doc_cp.sections:
+                gold_ents = []
+                for ent in section.metadata.get("gold_entities", []):
+                    if isinstance(ent, dict):
+                        ent = Entity.from_dict(ent)
+                    gold_ents.append(ent)
+                section.entities = gold_ents
+            result.append([doc_cp, doc])
+        return result
+
+    def update(
+        self, test_docs: list[Document], global_step: Union[int, str], has_gs: bool = True
+    ) -> None:
+        ls_manager = LabelStudioManager(
+            headers=self.ls_manager.headers,
+            project_name=f"{self.ls_manager.project_name}_test_{global_step}",
+        )
+
+        ls_manager.delete_project_if_exists()
+        ls_manager.create_linking_project()
+        docs_subset = random.sample(test_docs, min([len(test_docs), 100]))
```

Comment:
May add option to define the seed to enable reproduction of results?

Reply:
Yes you are correct, I think this should be in fact optional to only view a subset

Reply:
done